### PR TITLE
Add integration with gemlite weight only quant

### PR DESCRIPTION
**Pull request by jerryzh168**
Summary:
gemlite Only available with nightly torchao right now (or install from source)

Test Plan:
```
python3 -m sglang.bench_one_batch --model meta-llama/Llama-3.1-8B-Instruct --batch-size 1 --input 1024 --output 512 --json-model-override-args '{"architectures": ["TorchNativeLlamaForCausalLM"]}' --enable-torch-compile —torchao-config gemlite-4-64 --tp-size 1
```

Reviewers:

Subscribers:

Tasks:

Tags:

## Checklist

- [ ] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [ ] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [ ] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

**Comment by zhyncs** [comment]
Hi @jerryzh168 What is the release cycle of torchao? I can accept using the torchao nightly version, maybe you can try enabling it in the https://github.com/sgl-project/sglang/blob/main/python/pyproject.toml. What do you think? cc @merrymercy @Ying1123 @ispobock

File: python/sglang/srt/layers/torchao_utils.py [hunk 1]
```diff
@@ -47,6 +47,32 @@ def filter_fn(module, fqn):
...
+            # if only 2 inputs found, use default value
+            packing_bitwidth = 32
+
+        quantize_(model, gemlite_uintx_weight_only(group_size, bit_width, packing_bitwidth))
+
+        # try to load gemlite kernel config
+        try:
+            GemLiteLinearTriton.load_config(f"/tmp/{pwd.getpwuid(os.getuid()).pw_gecos}_gemlite.json")
+            print(f"loaded gemlite kernel cache /tmp/{pwd.getpwuid(os.getuid()).pw_gecos}_gemlite.json")
+        except:
```
Comment (jerryzh168) [inline python/sglang/srt/layers/torchao_utils.py]: cc @mobicham why do we need this try except? seems that GemLiteLinearTriton is already swallowing errors: https://github.com/mobiusml/gemlite/blob/5ebcca036943e7b1929f9be4beeeb0cf8d3fd170/gemlite/core.py#L427 (it probably should not do that I feel)

**Comment by jerryzh168** [comment]
> Hi @jerryzh168 What is the release cycle of torchao? I can accept using the torchao nightly version, maybe you can try enabling it in the [`main`/python/pyproject.toml](https://github.com/sgl-project/sglang/blob/main/python/pyproject.toml). What do you think? cc @merrymercy @Ying1123 @ispobock

we have ~ monthly releases, yeah depend on nightly version would be better for now, and we can update to a stable version a bit later I think

**Comment by jerryzh168** [comment]
I tried `pip install torchao>=0.8.0.dev20241219` but it doesn't work, we probably need to use
```
pip install --pre torchao --index-url https://download.pytorch.org/whl/nightly/cu124
```
to install the nightly version, do we just want to add a version check here?

**Comment by jerryzh168** [comment]
@zhyncs I think we can land, it's fine to have this as an experimental feature for now I think, I added a print to ask people to use torchao nightly

**Review by zhyncs — APPROVED** [review]

**Pull request merged**